src/ray/core_worker/store_provider/memory_store/memory_store.cc

```diff
@@ -103,6 +103,7 @@ bool GetRequest::Wait(int64_t timeout_ms) {
   auto remaining_timeout_ms = timeout_ms;
   auto timeout_timestamp = current_time_ms() + timeout_ms;
   while (!is_ready_) {
+    // TODO (dayshah): see if using cv condition function instead of busy while helps.
```

Comment:
ya i think it could still be relevant, pretty sure using ```cv.wait_for(lock, timeout, []() { return !is_ready; });``` could lead to less cpu usage vs the busy while loop but want to look more into how it would affect performance

Reply:
ok, let's do it in another PR, if needed